rllib/algorithms/impala/torch/vtrace_torch_v2.py

```diff
@@ -113,25 +113,6 @@ def vtrace_torch(
             on rho_s in \rho_s \delta log \pi(a|x) (r + \gamma v_{s+1} - V(x_s)).
     """
     log_rhos = target_action_log_probs - behaviour_action_log_probs
-    discounts = convert_to_torch_tensor(discounts)
```

Comment:
These seemed to be all superfluous. We are now only performing the for-loop on the CPU. Everything else (even within the vtrace_v2 functions) remains on the GPU.

Reply:
the dimension checking was useful for debugging. Is there some way we can incorporate this into the tests

Reply:
or would you entirely recommend against it all together (not worth the time)

Reply:
Agree, dimension checking is good, but we should not do it this "randomly" in the loss function. Instead, we should implement the same Spec checking mechanism that we have in the Modules now for the incoming nested dicts (`batch` and `fwd_out` that go into `compute_loss_per_module`). So I'm arguing for a more structured, unified way to pre-check incoming shapes and dtypes across RLlib's loss functions.

Reply:
lets add a todo then somehwere? or an issue. Just somewhere we can keep track of this.